### PR TITLE
Fixes #13661: Allow adding file type repositories to content views

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by ehelms** [comment]
Depends on https://github.com/Katello/katello/pull/5765

**Comment by ehelms** [comment]
Updated to be rebased on master

**Comment by jlsherrill** [comment]
Looks good and worked well, please rebase and i'll do one final test before an ACK

**Comment by ehelms** [comment]
Updated

**Comment by jlsherrill** [comment]
@ehelms retested and I'm not sure if i just didn't notice this before, but after publishing a content view with a file repo the resulting repo in the content view seems empty

**Comment by ehelms** [comment]
Now units are copying on publish and promote but will require https://github.com/Katello/runcible/pull/179

**Comment by ehelms** [comment]
[test]

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/content-views/details/views/content-view-file-repositories.html [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+          <th bst-table-column translate>Product</th>
+          <th bst-table-column translate>Last Sync</th>
+          <th bst-table-column translate>Sync State</th>
+        </tr>
+      </thead>
+
+      <tbody>
+        <tr bst-table-row
+            row-select="repository"
+            ng-repeat="repository in repositoriesTable.rows | filter:repositorySearch | filter:repositoryFilter | as:'filteredItems'">
```
Comment (jlsherrill) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/content-views/details/views/content-view-file-repositories.html]: This no longer works in master, see:  https://github.com/Katello/katello/pull/6212/files

File: app/lib/actions/katello/repository/clone_to_version.rb [hunk 2]
```diff
@@ -20,6 +20,8 @@ def plan(repository, content_view_version, incremental = false)
               plan_action(Repository::CloneDockerContent, repository, new_repository)
             elsif new_repository.ostree?
               plan_action(Repository::CloneOstreeContent, repository, new_repository)
+            elsif new_repository.file?
+              plan_action(Repository::CloneFileContent, repository, new_repository)
```
Comment (jlsherrill) [inline app/lib/actions/katello/repository/clone_to_version.rb]: I think you need this logic here too:  https://github.com/Katello/katello/blob/master/app/lib/actions/katello/repository/clone_to_environment.rb#L23

Comment (jlsherrill) [inline app/lib/actions/katello/repository/clone_to_version.rb]: This seems to be the reason why you weren't seeing directories created for the lifecycle environments.  After adding this logic to that file it seems to publish those directories.

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/content-views/details/content-view-available-file-repositories.controller.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+(function () {
```
Comment (jlsherrill) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/content-views/details/content-view-available-file-repositories.controller.js]: Tests for these js controllers?

**Comment by jlsherrill** [comment]
After those two changes, seems to work well

**Comment by ehelms** [comment]
@jlsherrill addressed all of your comments

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/content-views/details/views/content-view-file-repositories.html (hunk 1 above)
Comment (jlsherrill) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/content-views/details/views/content-view-file-repositories.html]: You update to fix this didn't work:
This was walden's updates:
 +            ng-repeat="repository in repositoriesTable.rows | filter:repositorySearch | filter:repositoryFilter as filteredItems"> 

and this was yours:

 +            ng-repeat="repository in repositoriesTable.rows | filter:repositorySearch | filter:repositoryFilter | as filteredItems">

**Comment by ehelms** [comment]
@jlsherrill subtle - updated

**Comment by jlsherrill** [comment]
undefined local variable or method `new_repository' for #<Actions::Katello::Repository::CloneToEnvironment:0x007f2822636d10>
when publishing a Content view

**Comment by jlsherrill** [comment]
hah i got the same problem when i tested this on friday.  Copy paste fail

**Comment by ehelms** [comment]
Volley

**Comment by jlsherrill** [comment]
APJ

**Pull request merged**